src/pages/Integrations/SplunkSetup/useSplunkSetup.tsx

```diff
@@ -74,52 +49,58 @@ const DEFAULT_SPLUNK_EVENTS: SplunkEventsDef = {
 };
 
 export const useSplunkSetup = () => {
+  const getRhelBundleUuid = useGetRhelBundleUuid();
   const createSplunkIntegration = useCreateSplunkIntegration();
-  const createSplunkBehaviorGroup = useCreateSplunkBehaviorGroup();
-  const updateSplunkBehaviorActions = useUpdateSplunkBehaviorActions();
-  const attachEvents = useAttachEventsToSplunk();
+  const attachEventTypes = useFindEventTypesToAssociate();
 
   return async ({ hecToken, splunkServerHostName }, onProgress) => {
     const now = new Date();
     const timestamp = now.getTime();
     const integrationName = `${SPLUNK_INTEGRATION_NAME}_${timestamp}`;
-    const behaviorGroupName = `${SPLUNK_BEHAVIOR_GROUP_NAME}_${timestamp}`;
-    const bundleName = BUNDLE_NAME;
     const events = DEFAULT_SPLUNK_EVENTS;
 
+    onProgress(`Fetching Event types...\n`);
+
+    const rhelBundleId = getRhelBundleUuid();
+
+    const eventTypeList = await attachEventTypes(
+      rhelBundleId,
+      events,
+      onProgress
+    );
+
     onProgress(`Creating Integration ${integrationName}...`);
-    const integration = await createSplunkIntegration({
+    await createSplunkIntegration({
       integrationName,
       hecToken,
       splunkServerHostName,
+      eventTypeList,
     });
-    onProgress(' OK', 'pf-v5-u-success-color-200');
-
-    onProgress(`\nCreating Behavior Group ${behaviorGroupName}...`);
-    const behaviorGroup = await createSplunkBehaviorGroup({
-      behaviorGroupName,
-      bundleName,
-    });
-    onProgress(' OK', 'pf-v5-u-success-color-200');
-
-    onProgress(
-      '\nAssociating integration as an action for the behavior group...'
-    );
-    await updateSplunkBehaviorActions(behaviorGroup, integration);
+  };
+};
 
-    onProgress(' OK', 'pf-v5-u-success-color-200');
-    onProgress('\n\nAssociating events to the behavior group:\n');
+const useGetRhelBundleUuid = () => {
+  const getBundles = useGetBundles();
 
-    await attachEvents(behaviorGroup, events, onProgress);
+  const getbundleTabs = () => {
+    if (getBundles.payload?.status === 200) {
```

Comment:
Isn't the `getBundles` an async function that needs to be called?

Reply:
Well I'm not sure, it is not used as async in other places of this project, so I used the same way to use it to be consistent.

Reply:
Is this the imported function: https://github.com/RedHatInsights/notifications-frontend/blob/23f50fc5f010c2abdbf844a78ffcef7154a411d6/src/services/Notifications/GetBundles.ts#L10-L13 ? Can you please double check?

Reply:
I feel like there is a incorrect wiring of these hooks. The ones that are created in this module do call async functions, and are hooks due to the frameworks that they take request capabilities from. The `useGetBundle` would create a hook [`useQuery` of react-fetching-library](https://marcin-piela.github.io/react-fetching-library/#/?id=usequery). The useQuery is set to issue the request immediately when the hook is created, triggering a rerender. Should the async function of `useSplunkSetup` was called immedeately or before the `useGetBundles` finishes loading, it would fail with the "error fetching bundle ...".

I recommend to set `initFetch = false` and then use `query` variable to manually call async request at the right time. See https://marcin-piela.github.io/react-fetching-library/#/?id=usequery

Reply:
Ok thank you, I updated this part, could you please have a look? BTW I also used an existing method to retrieve only one bundle.

Reply:
Thank you. It looks good. One minor thing, you might bring back a `onProgress()` before calling the async getBundle stuff to announce that step to the user.